services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/TeamManagerImpl.java

```diff
@@ -707,6 +707,28 @@ public String getIconURL(UserInfo userInfo, String teamId) throws NotFoundExcept
 		return fileHandleManager.getRedirectURLForFileHandle(urlRequest);
 	}
 
+	@Override
+	public String getIconPreviewURL(UserInfo userInfo, String teamId) throws NotFoundException {
+		Team team = teamDAO.get(teamId);
+
+		String fileHandleId = team.getIcon();
+
+		if (fileHandleId == null) {
+			throw new NotFoundException("Team " + teamId + " has no icon file handle.");
+		}
+
+		String fileHandlePreviewId = fileHandleManager.getPreviewFileHandleId(fileHandleId);
```

Comment:
This might throw a NotFoundException for both a non existing file handle id, or for a missing preview. We can catch it here and rethrow a more useful message targeted to this request.